packages/wonder-blocks-labeled-field/src/components/__tests__/labeled-field.test.tsx

```diff
@@ -215,11 +230,240 @@ describe("LabeledField", () => {
+                    error={error}
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const labelEl = screen.getByText(label);
+            const descriptionEl = screen.getByText(description);
+            const fieldEl = screen.getByRole("textbox");
+            const errorEl = screen.getByText(error);
+
+            // Assert
+            expect(labelEl.id).toBe(`${id}-label`);
+            expect(descriptionEl.id).toBe(`${id}-description`);
+            expect(fieldEl.id).toBe(`${id}-field`);
+            expect(errorEl.id).toBe(`${id}-error`);
+        });
+
+        it("should autogenerate ids for the elements if the id prop is not provided", () => {
+            // Arrange
+            const label = "Label";
+            const description = "Description of the field";
+            const error = "Error message";
+            render(
+                <LabeledField
+                    field={<TextField value="" onChange={() => {}} />}
+                    label={label}
+                    description={description}
+                    error={error}
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const labelEl = screen.getByText(label);
+            const descriptionEl = screen.getByText(description);
+            const fieldEl = screen.getByRole("textbox");
+            const errorEl = screen.getByText(error);
+
+            // Assert
+            expect(labelEl.id).toInclude("-label");
+            expect(descriptionEl.id).toInclude("-description");
+            expect(fieldEl.id).toInclude("-field");
+            expect(errorEl.id).toInclude("-error");
+        });
+
+        it("should use the testId prop for the base of test ids", () => {
+            // Arrange
+            const testId = "test-id";
+            const label = "Label";
+            const description = "Description of the field";
+            const error = "Error message";
+            render(
+                <LabeledField
+                    field={<TextField value="" onChange={() => {}} />}
+                    testId={testId}
+                    label={label}
+                    description={description}
+                    error={error}
+                />,
+                defaultOptions,
+            );
+
+            // Act
+            const labelEl = screen.getByText(label);
+            const descriptionEl = screen.getByText(description);
+            const fieldEl = screen.getByRole("textbox");
+            const errorEl = screen.getByText(error);
+
+            // Assert
+            expect(labelEl).toHaveAttribute("data-testid", `${testId}-label`);
+            expect(descriptionEl).toHaveAttribute(
+                "data-testid",
+                `${testId}-description`,
+            );
+            expect(fieldEl).toHaveAttribute("data-testid", `${testId}-field`);
+            expect(errorEl).toHaveAttribute("data-testid", `${testId}-error`);
```

Comment:
**suggestion:** This looks to me that we are testing multiple assertions within the same test. This makes a bit harder to understand when a specific assertion is failing. I think we should try using `it.each` [^1] to test every element within the component, instead of checking for multiple things in these tests.

Note: This comment would applly to all the tests that are asserting multiple things in this file.

[^1]: https://jestjs.io/docs/api#testeachtablename-fn-timeout

Reply:
Updated in https://github.com/Khan/wonder-blocks/pull/2339/commits/eaf4aa1edf3c68bf82a4ba6977f32b51ea12f48e! Let me know what you think :)

Reply:
Looks awesome! Neat approach for the `getElement` function.